### ISSUE TITLE
usb is closed ............?

### ISSUE THREAD

**Issue by iqbalpalemad** [rocketscream/Low-Power#31]
Hello 

i have uploaded the idleWakePeriodic example on my  BlueDuino . which has a ATMega32U4 processor.
i have commented the  line for ATMega32U4 in the example and uploaded the sketch. the sketch was uploaded successfully. But after that my pc is not recognizing the board. Which shows usb device not recognized . i think the usb close command is closed the usb function of board . so now how can i get back to my previous stage. now i cant upload any sketch...........?

**Comment by neolao** [comment]
When you press the reset button, you have a short time during which you can upload sketch

**Comment by iqbalpalemad** [comment]
but the usb device is not detecting to upload sketch

**Comment by iqbalpalemad** [comment]
have any other method.
i am un able to upload sketch. the port is not shown in arduino IDE

**Comment by neolao** [comment]
you can set the port manually in file preferences.txt

**Comment by iqbalpalemad** [comment]
can you please explain how to do it...?

the device is not shown in device manager

**Comment by neolao** [comment]
What is your system?

**Comment by iqbalpalemad** [comment]
Lenovo G510 with windows 10 os
i am using arduino IDE 1.6.2
the board used is BlueDuino

**Comment by iqbalpalemad** [comment]
Lenovo G510 with windows 10 os
i am using arduino IDE 1.6.12
the board used is BlueDuino

**Comment by neolao** [comment]
Ah ok, I use MacOS and Linux.

I think on Windows, you just have to find the right path to you physical USB port (the path you could select on IDE), maybe `COME6`.

![port](https://cloud.githubusercontent.com/assets/149506/21224518/d9b9f4aa-c2cc-11e6-8862-b7609867a584.png)

Close the IDE and you set in the file found here:

![preferences](https://www.kanda.com/blog/wp-content/uploads/arduino-preferences1.jpg)

**Comment by iqbalpalemad** [comment]
![screenshot 39](https://cloud.githubusercontent.com/assets/22958585/21224755/af74c89c-c2f3-11e6-8bfe-5d06ba04b8fc.png)


on my device manager the board is shown like this 

![screenshot 40](https://cloud.githubusercontent.com/assets/22958585/21224781/cd854708-c2f3-11e6-8c37-40b4f72fee22.png)


and no port is shown in arduino ide

**Comment by neolao** [comment]
Do you remember what was the port?

My solution is to not use the IDE to select the port.

**Comment by iqbalpalemad** [comment]
the port was com6

**Comment by rocketscream** [comment]
Always add a significant amount of delay at the start of the sketch to allow you the chance to upload code subsequently. ATMega32U4 still lacks a proper USB.detach() function call that supposed to be call before sleep mode is entered.
